File: solutions/0020-valid-parentheses/solution.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        int n = s.length();
        if(n == 0) return true;
        if(s.charAt(0) == '}' || s.charAt(0) == ']' || s.charAt(0) == ')') return false;
        
        char[] array = s.toCharArray();
        Stack<Character> stack = new Stack<Character>();
        
        for(char c: array) {
            if(c == '{' || c == '[' || c == '(') stack.push(c);
            
            else if(!stack.isEmpty()) {
                if(stack.peek() == '{' && c == '}') stack.pop();
                else if(stack.peek() == '[' && c == ']') stack.pop();
                else if(stack.peek() == '(' && c == ')') stack.pop();
                else return false;
            }
            
            else return false;
        }
        
        if(stack.isEmpty()) return true;
        else return false;
    }
    
    
}

File: solutions/0020-valid-parentheses/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    let stack = []
    for (let i = 0; i < s.length; i++) {
        if (['(', '{', '['].includes(s.charAt(i))) {
            stack.unshift(s.charAt(i))
        } else if ((')' === s.charAt(i) && stack[0] === '(')
            || ('}' === s.charAt(i) && stack[0] === '{')
            || (']' === s.charAt(i) && stack[0] === '[')) {
            stack.shift()
        } else {
            return false
        }
    }

    return stack.length === 0
};
